plasmapy/physics/transport.py

```diff
@@ -1,17 +1,78 @@
-from ..atomic import (ion_mass, charge_state)
-from .parameters import Debye_length
-
+from ..constants import (m_p,
+                         m_e,
+                         c,
+                         mu0,
+                         k_B,
+                         e,
+                         eps0,
+                         pi,
+                         h,
+                         hbar,
+                         )
+from ..atomic import (ion_mass,
+                      charge_state,
+                      )
+from .parameters import (Debye_length,
+                         )
+from .quantum import (Wigner_Seitz_radius,
+                      thermal_deBroglie_wavelength,
+                      chemical_potential,
+                      )
+from ..math import Fermi_integral
+
+
+
+
+def boilerPlate(T, particles, V):
+    """
+    Some boiler plate code for checking if inputs to functions in
+    transport.py are good. Also obtains reduced in mass in a
+    2 particle collision system along with thermal velocity.
+    """
+    # checking temperature is in correct units
+    T = T.to(u.K, equivalencies=u.temperature_energy())
+    # extracting particle information
+    if not isinstance(particles, (list, tuple)) or len(particles) != 2:
+        raise ValueError("Particles input must be a "
+                         "list or tuple containing representations of two  "
+                         f"charged particles. Got {particles} instead.")
+    masses = np.zeros(2) * u.kg
+    charges = np.zeros(2) * u.C
+    for particle, i in zip(particles, range(2)):
+        try:
+            masses[i] = ion_mass(particles[i])
+        except Exception:
+            raise ValueError("Unable to find mass of particle: "
+                             f"{particles[i]}.")
+        try:
+            charges[i] = np.abs(e * charge_state(particles[i]))
+            if charges[i] is None:
+                raise
+        except Exception:
+            raise ValueError("Unable to find charge of particle: "
+                             f"{particles[i]}.")
+    # obtaining reduced mass of 2 particle collision system
+    reduced_mass = masses[0] * masses[1] / (masses[0] + masses[1])
+    # getting thermal velocity of system if no velocity is given
+    if V is None:
+        V = np.sqrt(3 * k_B * T / reduced_mass)
+    else:
+        _check_relativistic(V, 'impact_parameter', betafrac=0.8)
+    return (T, masses, charges, reduced_mass, V)
 
-@check_quantity({"T": {"units": units.K, "can_be_negative": False},
-                 "n_e": {"units": units.m**-3}
+@check_quantity({"T": {"units": u.K, "can_be_negative": False},
+                 "n_e": {"units": u.m**-3}
                  })
-def Coulomb_logarithm(T, n_e, particles, V=None):
+def Coulomb_logarithm(T,
+                      n_e,
+                      particles,
+                      V=None,
+                      method="classical"):
     r"""Estimates the Coulomb logarithm.
 
     Parameters
```

Comment:
We'll definitely need an update to docs here, with the new arguments and so on. But of course that's a larger-than-usual endeavour.

Reply:
Yes... perhaps once I've finished the coding you guys can give me some input on making this clear and understandable to an audience who has not been exposed to these concepts 😅

Reply:
Sure! :laughing: